src/fastpath/fastpath.c

```diff
@@ -99,7 +99,23 @@ void NORETURN fastpath_call(word_t cptr, word_t msgInfo)
     stored_hw_asid.words[0] = 0;
 #endif
 #ifdef CONFIG_ARCH_AARCH64
-    stored_hw_asid.words[0] = cap_vtable_root_get_mappedASID(newVTable);
+    /* Need to test that the ASID is still valid */
+    asid_t asid = cap_vtable_root_get_mappedASID(newVTable);
+    asid_map_t asid_map = findMapForASID(asid);
+    if (unlikely(asid_map_get_type(asid_map) != asid_map_asid_map_vspace ||
+                 VSPACE_PTR(asid_map_asid_map_vspace_get_vspace_root(asid_map)) != cap_pd)) {
+        slowpath(SysCall);
+    }
+#ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
+    /* Ensure the HWASID is valid. */
+    if (unlikely(!asid_map_asid_map_vspace_get_stored_vmid_valid(asid_map))) {
+        slowpath(SysCall);
```

Comment:
```suggestion
    /* Ensure the vmid is valid. */
    if (unlikely(!asid_map_asid_map_vspace_get_stored_vmid_valid(asid_map))) {
        slowpath(SysCall);
```

Reply:
Given the difference in naming in the other configs (`hw_asid`), might need a comment that `vmid`s are the tags used in hyp mode.

Reply:
I went with the vmid naming because of the discussion on the draft PR, but to be more complete, all the other function names should probably be changed too...

Reply:
Yes, I agree, was just about to write that in a summary comment. Might make sense do that rename in a separate commit/PR.

Reply:
Probably even the type `hw_asid_t`, if that doesn't conflict with code that is shared between configs (might be a problem in the fastpath for instance).